packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -128,10 +128,21 @@ export default createRule({
 
           caseTypes.add(getNodeType(switchCase.test));
         }
+        const missingBranchTypes = unionTypes.filter(unionType => {
+          const type = isTypeFlagSet(
+            unionType,
+            ts.TypeFlags.Number |
+              ts.TypeFlags.String |
+              ts.TypeFlags.Object |
+              ts.TypeFlags.Intersection,
+          );
```

Comment:
🤔 maybe this is because the variable name `type` isn't super descriptive, but I'm finding it hard to understand what this check does. Could you explain how this change fixes the issue / new test cases?

Taking a step back, I think what we really care about is whether any of the union type constituents are an infinite type. Is there a more straightforward way to codify that?

Reply:
I believe `type` indicates whether this union constituent is infinite? Doesn't look correct though, because infinite types can be practically anything, like symbols etc... May be easier to do something like `unionTypes.every(isLiteralType)`